common/models/role-mapping.js

```diff
@@ -26,9 +26,9 @@ module.exports = function(RoleMapping) {
   RoleMapping.resolveRelatedModels = function() {
     if (!this.userModel) {
       var reg = this.registry;
-      this.roleModel = reg.getModelByType(loopback.Role);
-      this.userModel = reg.getModelByType(loopback.User);
-      this.applicationModel = reg.getModelByType(loopback.Application);
+      this.roleModel = reg.getModelByType('Role');
+      this.userModel = reg.getModelByType('User');
+      this.applicationModel = reg.getModelByType('Application');
     }
```

Comment:
using `reg.getModelByType(loopback.Role)` ; caused the following tests with principal.user() to fail, whereas using `reg.getModelByType('Role')` solved the issue. This is related to the new one-register-per-app recent change.
I think doing similar changes for loopback.User and loopback.Application is safer